InnerEye/ML/configs/segmentation/HeadAndNeckPaper.py

```diff
@@ -0,0 +1,74 @@
+from InnerEye.ML.utils.model_metadata_util import generate_random_colours_list
+from .HeadAndNeckBase import HeadAndNeckBase
+
+# List of structures to segment. The order is important, because different values of num_structures
+# in the constructor will select different prefixes of the list.
+
+STRUCTURE_LIST = ["external", "parotid_l", "parotid_r", "smg_l", "smg_r", "spinal_cord", "brainstem",
+                  "globe_l", "globe_r", "mandible", "spc_muscle", "mpc_muscle", "cochlea_l", "cochlea_r",
+                  "lens_l", "lens_r", "optic_chiasm", "optic_nerve_l", "optic_nerve_r", "pituitary_gland",
+                  "lacrimal_gland_l", "lacrimal_gland_r"]
+RANDOM_COLOUR_GENERATOR = random.Random(0)
+COLOURS = generate_random_colours_list(RANDOM_COLOUR_GENERATOR, len(STRUCTURE_LIST))
+
+class HeadAndNeckPaper(HeadAndNeckBase):
+    '''
+    Head and Neck model, as used in the paper.
+    '''
+
+    def __init__(self, num_structures: Optional[int] = None, **kwargs: Any) -> None:
+        '''
+        Creates a new instance of the class.
+        :param num_structures: number of structures from STRUCTURE_LIST to predict (default: all structures)
+        :param kwargs: Additional arguments that will be passed through to the SegmentationModelBase constructor.
+        '''
+        # Number of structures to predict; if positive but less than the length of STRUCTURE_LIST, the relevant prefix
+        # of STRUCTURE_LIST will be predicted.
+        if (num_structures is None) or num_structures <= 0 or num_structures > len(STRUCTURE_LIST):
+            num_structures = len(STRUCTURE_LIST)
+        ground_truth_ids = STRUCTURE_LIST[:num_structures]
+        ground_truth_ids_display_names = kwargs.pop("ground_truth_ids_display_names", [f"zz_{x}" for x in ground_truth_ids])
+        colours = kwargs.pop("colours", COLOURS[:num_structures])
+        fill_holes = kwargs.pop("fill_holes", [True] * num_structures)
+        # The amount of GPU memory required increases with both the number of structures and the
+        # number of feature channels. The following is a sensible default to avoid out-of-memory,
+        # but you can override is by passing in another (singleton list) value for feature_channels
+        # from a subclass.
+        num_feature_channels = kwargs.pop("num_feature_channels", 32 if num_structures <= 20 else 26)
+        bg_weight = 0.02 if len(ground_truth_ids) > 1 else 0.25
+        class_weights = kwargs.pop("class_weights", equally_weighted_classes(ground_truth_ids, background_weight=bg_weight))
+        # In case of vertical overlap between brainstem and spinal_cord, we separate them
+        # by converting brainstem voxels to cord, as the latter is clinically more sensitive.
+        # We do the same to separate SPC and MPC; in this case, the direction of change is unimportant,
+        # so we choose SPC-to-MPC arbitrarily.
+        slice_exclusion_rules = []
```

Comment:
I would apply the same logic with kwargs.pop here. If they are set explicitly, use them, otherwise branch into the logic here. Also, logging.info should say that these are added automagically.